datadog/resource_datadog_powerpack.go

```diff
@@ -707,37 +735,113 @@ func normalizeDashboardWidgetDef(widgetDef map[string]interface{}, columnWidth i
 	}
 
 	if widgetDef["request"] != nil {
-		castWidgetDefReq := *widgetDef["request"].(*[]map[string]interface{})
-		if len(castWidgetDefReq) == 0 {
-			diags = append(diags, diag.Diagnostic{
-				Severity: diag.Error,
-				Summary:  fmt.Sprintf("at least one request should be defined for widget: %s", widgetDef["type"]),
-			})
-			return nil, diags
+		if widgetDef["type"] == "scatterplot" || widgetDef["type"] == "hostmap" {
+			// Scatterplot and hostmap widgets expect a single requests object instead of a list
+			widgetDefRequest := widgetDef["request"].([]map[string]interface{})[0]
+			if len(widgetDefRequest) == 0 {
+				diags = append(diags, diag.Diagnostic{
+					Severity: diag.Error,
+					Summary:  fmt.Sprintf("request should be defined for widget: %s", widgetDef["type"]),
+				})
+				return nil, diags
+			}
+			for _, v := range []string{"fill", "size", "x", "y"} {
+				// Properties listed above are always a single value, not a list
+				if widgetDefRequest[v] != nil {
+					widgetDefRequest[v] = widgetDefRequest[v].([]map[string]interface{})[0]
+				}
+			}
+			widgetDef["requests"] = widgetDefRequest
+		} else {
+			castWidgetDefReq := *widgetDef["request"].(*[]map[string]interface{})
+			if len(castWidgetDefReq) == 0 {
+				diags = append(diags, diag.Diagnostic{
+					Severity: diag.Error,
+					Summary:  fmt.Sprintf("at least one request should be defined for widget: %s", widgetDef["type"]),
+				})
+				return nil, diags
+			}
+			// Distribution/change/heatmap widgets have a "requests" field, while API Spec has a "request" field
+			// Here we set the "requests" field and remove "request"
+			widgetDef["requests"] = normalizeWidgetDefRequests(castWidgetDefReq, widgetDef["type"].(string))
 		}
-		// Distribution/change/heatmap widgets have a "requests" field, while API Spec has a "request" field
-		// Here we set the "requests" field and remove "request"
-		widgetDef["requests"] = castWidgetDefReq
 		delete(widgetDef, "request")
 	}
+	for _, v := range []string{"style", "xaxis", "yaxis"} {
+		// Properties listed above are always a single value, not a list
+		if widgetDef[v] != nil {
+			widgetDef[v] = widgetDef[v].([]map[string]interface{})[0]
+		}
+	}
+	for _, v := range []string{"event", "custom_link", "input"} {
+		// Properties listed above are defined as single, but API Spec expects a plural name
+		if widgetDef[v] != nil {
+			widgetDef[v+"s"] = widgetDef[v]
+			delete(widgetDef, v)
+		}
+	}
```

Comment:
**suggestion(non-blocking):** Similar to a suggestion above, we may want to consider remapping properties w/ some kinda map here, rather than relying the on the "s"-suffix relation